group-coordinator/src/main/resources/common/message/ConsumerGroupCurrentMemberAssignmentValue.json

```diff
@@ -22,16 +22,29 @@
   "fields": [
     { "name": "MemberEpoch", "versions": "0+", "type": "int32",
       "about": "The member epoch." },
+    { "name": "PreviousMemberEpoch", "versions": "0+", "type": "int32",
+      "about": "The previous member epoch." },
+    { "name": "TargetMemberEpoch", "versions": "0+", "type": "int32",
+      "about": "The target member epoch." },
+    { "name": "AssignedPartitions", "versions": "0+", "type": "[]TopicPartitions",
+      "about": "The partitions assigned to (owned by) this member." },
+    { "name": "PartitionsPendingRevocation", "versions": "0+", "type": "[]TopicPartitions",
+      "about": "The partitions being revoked by this member." },
+    { "name": "PartitionsPendingAssignment", "versions": "0+", "type": "[]TopicPartitions",
+      "about": "The partitions being assigned to this member." },
```

Comment:
This is the biggest divergence from the KIP. Initially, I wanted to store only the full assignment and to bookkeep those in-memory. I found this extremely confusing during the implementation and error prone as well. It is much better to have three sets instead of ones.